src/ai/defaultai.cc

```diff
@@ -784,7 +784,7 @@ void DefaultAI::late_initialization() {
 			}
 
 			for (const auto& temp_position : prod.working_positions()) {
-				for (uint8_t i = 0; i < temp_position.second; i++) {
+				for (time_t i = 0; i < temp_position.second; i++) {
```

Comment:
What I did here is obviously wrong!

'second' appears to be the second member of a pair struct. I read a bit about pairs... it seems that the type is determined at compile time.

How do I declare 'i' correctly here since it's not known until compile-time?

Reply:
second position of the pair is the amount of the worker. it is defined in logic/map_objects/tribes/productionsite.cc in line 56 as int, so declaring it as size_t might be good.     
  
However I deem the posible Multiplication overflows as mote critical as I had this case once in the statistics code calculating the percentage of productivity.